src/messaging/ExchangeContext.cpp

```diff
@@ -123,7 +124,15 @@ CHIP_ERROR ExchangeContext::SendMessageImpl(Protocols::Id protocolId, uint8_t ms
     // an error arising below. at the end, we have to close it.
     Retain();
 
-    bool reliableTransmissionRequested = !sendFlags.Has(SendMessageFlags::kNoAutoRequestAck);
+    // If sending via UDP and NoAutoRequestAck send flag is not specificed, request reliable transmission.
```

Comment:
Since this sets bools inside ifs, would this be readable enough?

```cpp
reliableTransmissionRequested =                                              //
    state &&                                                                 //
    (state->GetPeerAddress().GetTransportType() == Transport::Type::kUdp) && //
    !sendFlags.Has(SendMessageFlags::kNoAutoRequestAck);
```

Reply:
Declare reliableTransmissionRequested before ifs to make it more readable